File: src/main/java/com/adjorno/billib/rest/ChartController.java
package com.adjorno.billib.rest;

import com.adjorno.billib.rest.db.Chart;
import com.adjorno.billib.rest.db.ChartRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ChartController {
    @Autowired
    private ChartRepository mChartRepository;

    @RequestMapping(value = "/chart/all", method = RequestMethod.GET)
    public Iterable<Chart> getAllCharts() {
        return mChartRepository.findAll();
    }

}

File: src/main/java/com/adjorno/billib/rest/ChartController.kt
package com.adjorno.billib.rest

import com.adjorno.billib.rest.db.Chart
import org.springframework.web.bind.annotation.RestController
import org.springframework.beans.factory.annotation.Autowired
import com.adjorno.billib.rest.db.ChartRepository
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod

@RestController
class ChartController {
    @Autowired
    private lateinit var chartRepository: ChartRepository

    @get:RequestMapping(value = ["/chart/all"], method = [RequestMethod.GET])
    val allCharts: Iterable<Chart>
        get() = chartRepository.findAll()
}